### PR TITLE
Migrate organization_bootstrap command to ANONYMOUS_CMDS

### LINKED ISSUE #2196: Migrate organization_bootstrap command to ANONYMOUS_CMDS
The `organization_bootstrap` command is the only command still in use on the client side from the APIv1 (as part of `APIV1_ANONYMOUS_CMDS`). Migrating this command to the new HTTP api for anonymous command would allow to finally get rid of all the legacy code related to APIv1 (once all the clients have the new `organization_bootstrap` implementation).

### PR DESCRIPTION
**Pull request by vxgmichel**
Fix #2196

### CODE REVIEW

File: parsec/api/protocol/organization.py [hunk 1]
```diff
@@ -37,6 +37,32 @@ class APIV1_OrganizationBootstrapRepSchema(BaseRepSchema):
 )
 
 
+class OrganizationBootstrapReqSchema(BaseReqSchema):
+    bootstrap_token = fields.String(required=True)
+    root_verify_key = fields.VerifyKey(required=True)
+    user_certificate = fields.Bytes(required=True)
+    device_certificate = fields.Bytes(required=True)
+    # Same certificates than above, but expurged of human_handle/device_label
+    # Backward compatibility prevent those field to be required, however
```
Comment (touilleMan) [inline parsec/api/protocol/organization.py:46]: No need for backward compatibility here given this command is brand new !

File: parsec/backend/organization.py [hunk 2]
```diff
@@ -140,10 +142,20 @@ async def api_authenticated_organization_stats(self, client_ctx, msg):
...
+        # Use the correct serializer depending on the API
+        if isinstance(client_ctx, APIV1_AnonymousClientContext):
+            serializer = apiv1_organization_bootstrap_serializer
+        else:
+            assert isinstance(client_ctx, AnonymousClientContext)
+            serializer = organization_bootstrap_serializer
+
+        msg = serializer.req_load(msg)
+
         bootstrap_token = msg["bootstrap_token"]
```
Comment (touilleMan) [inline parsec/backend/organization.py:159]: should add a comment above where `if "redacted_user_certificate" in msg:` is done given this won't be needed once APIv1 is dropped

File: parsec/core/cli/bootstrap_organization.py [hunk 3]
```diff
@@ -32,22 +31,19 @@ async def _bootstrap_organization(
         device_label_raw = await aprompt("Device label", default=platform.node())
         device_label = DeviceLabel(device_label_raw)
 
-    async with apiv1_backend_anonymous_cmds_factory(addr=addr) as cmds:
-        async with spinner("Bootstrapping organization in the backend"):
-            new_device = await do_bootstrap_organization(
-                cmds=cmds, human_handle=human_handle, device_label=device_label
-            )
+    async with spinner("Bootstrapping organization in the backend"):
+        new_device = await do_bootstrap_organization(
```
Comment (touilleMan) [inline parsec/core/cli/bootstrap_organization.py:35]: sooo much better not to have to create a cmds ! \o/

**Review by touilleMan — APPROVED** [review]

**Review by touilleMan — APPROVED** [review]

File: parsec/backend/http.py [hunk 4]
```diff
@@ -367,33 +367,51 @@ async def _api_check_auth_and_load_body(self, req: HTTPRequest, req_serializer):
         return None, data
 
     async def _http_api_anonymous(self, req: HTTPRequest, **kwargs: str) -> HTTPResponse:
+        # Check whether the organization exists
```
Comment (touilleMan) [inline parsec/backend/http.py:370]: Are we sure this is what we want to do ?

On the one hand it's tempting to have bootstrap as part of the anonymous family given they all refer to an organization_id, but on the other hand the fact the organization should not be bootstrapped (and may even not be created at all) make it very special (hence those non-trivial changes that may be error prone...)

I don't have a strong opinion, but I think it's interesting to think a bit about that before choosing our poison ;-)

**Review by touilleMan — APPROVED** [review]

File: parsec/backend/http.py (hunk 4 above)
Comment (vxgmichel) [reply parsec/backend/http.py:370]: Agreed, as a first implementation I chose to follow what was already done here:
https://github.com/Scille/parsec-cloud/blob/f4a624d2c0152842f4cd5d82b510728d85c3428a/parsec/backend/handshake.py#L194-L235

But we might want to change that, I don't have a strong opinion either.

Comment (touilleMan) [reply parsec/backend/http.py:370]: > But we might want to change that, I don't have a strong opinion either.

So how do we decide if it's good or not ? :trollface:

Comment (vxgmichel) [reply parsec/backend/http.py:370]: > the fact the organization should not be bootstrapped (and may even not be created at all) make it very special (hence those non-trivial changes that may be error prone...)

The fact that the organization should not be bootsrapped is already managed correctly in the command itself. The weird part is the spontaneous creation of the organization if the backend configuration allows it.

Comment (vxgmichel) [reply parsec/backend/http.py:370]: Let's keep it this way for the moment and investigate later how we want to evolve this command.

**Pull request merged**